container/go/cmd/puller/puller.go

```diff
@@ -30,11 +31,13 @@ import (
 	"github.com/google/go-containerregistry/pkg/authn"
 	"github.com/google/go-containerregistry/pkg/name"
 	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
 )
 
 var (
 	imgName         = flag.String("name", "", "The name location including repo and digest/tag of the docker image to pull and save. Supports fully-qualified tag or digest references.")
-	directory       = flag.String("directory", "", "Where to save the images files.")
+	directory       = flag.String("directory", "", "Where to save the images files. If pulling as Docker tarball, please specify the directory to save the tarball. The tarball is named as image.tar.")
+	format          = flag.String("format", "", "The format, options being 'oci', docker', or 'both', in which to pull the docker image. 'docker' tarball may be used with `docker load -i`. Default 'oci', OCI layout format.")
```

Comment:
Make it explicitly that if format is docker, it's pulled as a tarball, if the format is oci, image will be pulled as a collection of files in OCI layout into a directory.

Also Include this description in the doc above after "This binary pulls images from a Docker Registry using the go-containerregistry as backend."